chaudloader/src/mods/lua/lib/chaudloader/pck.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{mods, path};
+use mlua::ExternalError;
+
+pub fn new<'a>(
+    lua: &'a mlua::Lua,
+    mod_path: &std::path::Path,
+) -> Result<mlua::Value<'a>, mlua::Error> {
+    let table = lua.create_table()?;
+
+    table.set(
+        "load_pck",
+        lua.create_function({
+            let mod_path = mod_path.to_path_buf();
+            move |_, (path,): (String,)| {
+                match path.as_str() {
+                    "Vol1.pck" | "Vol2.pck" | "DLC1.pck" | "DLC2.pck" | "chaudloader.pck" => {
+                        return Err(anyhow::anyhow!("cannot use the names: Vol1.pck, Vol2.pck, DLC1.pck, DLC2.pck, or chaudloader.pck").into_lua_err());
+                    },
+                    _ => {
+                        let path = path::ensure_safe(std::path::Path::new(&path))
+                            .ok_or_else(|| anyhow::anyhow!("cannot read files outside of mod directory"))
+                            .map_err(|e| e.into_lua_err())?;
+
+                        let pck_path = mod_path.join(&path);
+                        if !pck_path.exists() {
+                            return Err(anyhow::anyhow!("{} does not exist", pck_path.display()).into_lua_err());
+                        }
+                        let mut mod_audio = mods::MODAUDIOFILES.get().unwrap().lock().unwrap();
+                        // Check if a pck with this file name is already being loaded because the pck load function only takes the base file name.
+                        let base_filename = pck_path.file_name().unwrap().to_os_string();
+                        if mod_audio.pcks.contains(&base_filename) {
+                            return Err(anyhow::anyhow!("a pck file named {} is already being loaded", base_filename.to_str().unwrap()).into_lua_err());
+                        } else {
+                            // Copy this pck to the audio folder so it can be loaded
```

Comment:
Is this really necessary? E.g. would feeding it a relative path not work?

Can we use the asset replacer here? (See `on_create_file`) I'd like to avoid copying mod files into game data folders if possible, since they aren't cleaned up when the mod is disabled/removed.